### ISSUE TITLE
Nameplate Mouseover Area Misalignment

### ISSUE THREAD

**Issue by dmcurtis1** [Backupiseasy/ThreatPlates#465]
The bounding box for nameplate mouseover is skewed to the right and doesn't line up with the "Clickable Area" setting. This issue persists even without any other addons enabled.
https://github.com/Backupiseasy/ThreatPlates/assets/63743560/fa5fb989-5da8-4a88-b0a7-f7ab6a8aad6a

**Comment by Backupiseasy** [comment]
That is interessting. The clickable area is - to my knowledge - always centered on the character. Even if the configuration area is wrong, the actual clickable area should never be off-center. To my knowledge, addons can only change the height/width of the clickable area - or more like the Blizzard nameplate which is hidden - but not change it's positioning. 

No idea why that should happen or how that is even possible. I will investigate, but I am not optimistic.

**Comment by Backupiseasy** [comment]
Is this Classic?

**Comment by dmcurtis1** [comment]
This is WotLk Classic. For the record I've witnessed the same behavior with both ThreatPlates and Plater. I can temporarily provoke the nameplate mouseover area to function normally with this sequence:

1. Navigate to: 
(/tptp, Blizzard Settings, "Names" tab) 
-->Check "Names Only" box, wait for automatic reload to complete

2. Navigate to:
(/tptp, Visibility)
-->Check "Show Blizzard Nameplates for Friendly Units"
-->Check "Show Blizzard Nameplates for Neutral and Enemy Units"

3. Navigate back to:
(/tptp, Blizzard Settings, "Names" tab)
-->Uncheck "Names Only" box, wait for automatic reload to complete

4. Navigate back to (/tptp, Visibility)
-->Uncheck "Show Blizzard Nameplates for Friendly Units"
-->Uncheck "Show Blizzard Nameplates for Neutral and Enemy Units"

For whatever reason, this sequence causes my Nameplate Mouseover area to now correctly obey the parameters set in the "Clickable Area" settings, and function exactly as expected. Perfect fix? Sort-of. The only problem now is that this fix doesn't persist. It has to be repeated after every loading screen, /reload, or relog.

p.s. Don't ask me how I figured out that sequence fixed things. I've been pulling my hair out for months trying to figure this bug out lmao

**Comment by dmcurtis1** [comment]
Any leads on this issue?

**Comment by Backupiseasy** [comment]
No, not yet. I (seem to) remember that I had a situation like that when testing changing the clickable area in Classic. That's why I asked if this is classic. But I cannot remember what I did back then :(

**Comment by Backupiseasy** [comment]
Can you both test something for me? If I change the clickable area, it looks like this:

![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/784f1b56-5674-47cf-ab07-48cc2ddb5e32)

As you can see, although healthbar and nameplate are not aligned, the healthbar is centered on the character. For you, it should look different, more like this:

![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/39be96ea-5b76-4745-9720-dadf05f5b0ef)
(Ignore the bad graphics editing!)

You can test that by disabling Threat Plates for friendly and/or enemy units in General - Visibility.

If you want to see if the off-center alignment increases with a wider nameplate, you can use this command:

`/script C_NamePlate.SetNamePlateFriendlySize(300, 64)`

How does the Blizzard nameplate (like above) look for you when the clickable area is off-center?

Also, as you mentioned that this happens with Plater and Threat Plates, does this also happen when you disable these addons and use default Blizzard plates? How does ist look? Is the visible nameplate/healthbar centered, but the clickable area is not?

**Comment by dmcurtis1** [comment]
Any updates on a solution to this bug now that the ICC patch has been live for awhile? The bug still exists in game.

**Comment by dmcurtis1** [comment]
This issue is still present in the Cata prepatch. It's also present in SOD.

**Comment by Backupiseasy** [comment]
I am sorry. I did work on this a bit, but then gut side-tracked. I will look into it. Your explaination is good enough to understand and hopefully to fix it. Thanks for your effort.

**Comment by Backupiseasy** [comment]
A small update: I did some tests and re-worked how the clickable area is calulated based on the healthbar size (if synced with it). Right now, that''s only finished for Mainline, but next is doing the same for Classic. 

Here's how the clickable area looks on Mainline for small and large Blizzard nameplates:

- Small nameplates (also, not that the smaller healthbar is not centered vertically like the large healthbar): 
![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/8845b87e-5096-4a34-8526-bd22c1ee6bbc)
- Large Nameplates
![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/48b1fc61-37b6-4242-b9a0-55a3fd900a84)

If you set the TP healthbar to the same size as the Blizzard nameplate (small: 88.5 x 4.3, large: 129.5 x 10.6), then the clickable area is the same now. The default size of TP nameplates is 120 x 10, so a bit smaller als large nameplates, so by default the clickable area is also a bit smaller. Maybe I should change that ...

- TP with small nameplates (with TP, the small healthbar is actually centered vertically):
![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/9b3c7c03-b395-4205-abb2-401d97c8cc6c)
- TP with large nameplates:
![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/fa67d1bd-d15e-478b-ba2c-9c52360ef263)

Next is doing the same for Classic :D

**Comment by Backupiseasy** [comment]
For Classic, it now looks like this:

- Blizzard nameplates
![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/42f947ff-a2b7-47bc-b559-f323f42928ed)
- Threat Plates of the same size (width is 122)
![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/a8cad6b2-7089-4b1c-ba05-78c94621b3e0)
- Threat Plates with a larger healthbar:
![image](https://github.com/Backupiseasy/ThreatPlates/assets/20885351/b74ccbed-ffa4-4170-84f4-1e8ed53bbdc5)

I uploaded a version for 

- Cata Classic: https://drive.google.com/file/d/1J0hglh9A6bskBlbQOOXzoN9PB_oPOVMk/view?usp=sharing, and for
- Classic: https://drive.google.com/file/d/1bavqATjl9fObxpHeRzvnIhF5PsPHt9t0/view?usp=sharing

for testing. With this version, the above issues should be gone. There should also be no mis-alignment anymore. 

You can show the clickable area with the chat command `/tptp clickable-area`

Feedback is highly appreciated! :D

**Comment by dmcurtis1** [comment]
The misalignment issue (the one demonstrated in the video of the original post) is still present in Cata Classic with Threat Plates v12.0.6.

**Comment by dmcurtis1** [comment]
> This is WotLK Classic. For the record I've witnessed the same behavior with both ThreatPlates and Plater. I can temporarily provoke the nameplate mouseover area to function normally with this sequence:
> 
> 1. Navigate to:
>    (/tptp, Blizzard Settings, "Names" tab)
>    -->Check "Names Only" box, wait for automatic reload to complete
> 2. Navigate to:
>    (/tptp, Visibility)
>    -->Check "Show Blizzard Nameplates for Friendly Units"
>    -->Check "Show Blizzard Nameplates for Neutral and Enemy Units"
> 3. Navigate back to:
>    (/tptp, Blizzard Settings, "Names" tab)
>    -->Uncheck "Names Only" box, wait for automatic reload to complete
> 4. Navigate back to:
>    (/tptp, Visibility)
>    -->Uncheck "Show Blizzard Nameplates for Friendly Units"
>    -->Uncheck "Show Blizzard Nameplates for Neutral and Enemy Units"
> 
> For whatever reason, this sequence causes my Nameplate Mouseover area to now correctly obey the parameters set in the "Clickable Area" settings, and function exactly as expected. Perfect fix? Sort-of. The only problem now is that this fix doesn't persist. It has to be repeated after every loading screen, /reload, or relog.
> 
> p.s. Don't ask me how I figured out that sequence fixed things. I've been pulling my hair out for months trying to figure this bug out lmao

I can still fix it using this sequence, but that fix is still temporary and is immediately reset after every loading screen or /reload.

**Comment by Backupiseasy** [comment]
When this mis-alignment is active, can you use the chat command /tptp clickable-area and post a screenshot here how it looks? I cannot reproduce this currently.

**Comment by Backupiseasy** [comment]
I completely reworked clickable area (in Classic and Mists Classic/Midnight). I cannot find any errors right now.
